meca:
  left_forward_controller:
    type: velocity_controllers/JointVelocityController
    joint: Left_forward_joint1
    publish_rate: 50
    pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    base_frame_id: base_footprint

    linear:
      x:
        has_velocity_limits: true
        max_velocity: 3  # Changed from 0 to 1 m/s
        has_acceleration_limits: true
        max_acceleration: 3  # m/s^2
    angular:
      z:
        has_velocity_limits: true
        max_velocity: 3  # rad/s
        has_acceleration_limits: true
        max_acceleration: 3  # rad/s^2

  left_backward_controller:
    type: velocity_controllers/JointVelocityController
    joint: Left_backward_joint1
    publish_rate: 50
    pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    base_frame_id: base_footprint

    linear:
      x:
        has_velocity_limits: true
        max_velocity: 3  # Changed from 0 to 1 m/s
        has_acceleration_limits: true
        max_acceleration: 3  # m/s^2
    angular:
      z:
        has_velocity_limits: true
        max_velocity: 3  # rad/s
        has_acceleration_limits: true
        max_acceleration: 3  # rad/s^2

  right_forward_controller:
    type: velocity_controllers/JointVelocityController
    joint: Right_forward_joint1
    publish_rate: 50
    pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    base_frame_id: base_footprint

    linear:
      x:
        has_velocity_limits: true
        max_velocity: 3  # m/s
        has_acceleration_limits: true
        max_acceleration: 3  # m/s^2
    angular:
      z:
        has_velocity_limits: true
        max_velocity: 3  # rad/s
        has_acceleration_limits: true
        max_acceleration: 3  # rad/s^2

  right_backward_controller:
    type: velocity_controllers/JointVelocityController
    joint: Right_backward_joint1
    publish_rate: 50
    pose_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    twist_covariance_diagonal: [0.001, 0.001, 0.001, 0.001, 0.001, 0.03]
    base_frame_id: base_footprint

    linear:
      x:
        has_velocity_limits: true
        max_velocity: 3  # m/s
        has_acceleration_limits: true
        max_acceleration: 3  # m/s^2
    angular:
      z:
        has_velocity_limits: true
        max_velocity: 3  # rad/s
        has_acceleration_limits: true
        max_acceleration: 3  # rad/s^2
        has_acceleration_limits: true
        max_acceleration: 4  # rad/s^2

